src/main/java/uk/gov/cshr/service/reportingService/ReportingService.java

```diff
@@ -10,36 +10,49 @@
 import uk.gov.cshr.service.RequestEntityFactory;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 
+import static java.util.Collections.singletonList;
+
 @Service
 @Slf4j
 public class ReportingService {
     private final HttpClient httpClient;
     private final RequestEntityFactory requestEntityFactory;
     private final String removeUserDetailsUrl;
+    private final String removeUserDataFromReportUrl;
 
     public ReportingService(
             HttpClient httpClient,
             RequestEntityFactory requestEntityFactory,
-            @Value("${reporting.removeUserDataUrl}") String removeUserDetailsUrl
+            @Value("${reporting.removeUserDataUrl}") String removeUserDetailsUrl,
+            @Value("${reporting.removeUserDataFromReportUrl}") String removeUserDataFromReportUrl
     ) {
         this.httpClient = httpClient;
         this.requestEntityFactory = requestEntityFactory;
         this.removeUserDetailsUrl = removeUserDetailsUrl;
+        this.removeUserDataFromReportUrl = removeUserDataFromReportUrl;
     }
 
-    public ResponseEntity<Void> removeUserDetails(List<String> uids){
+    public ResponseEntity<Void> removeUserDetails(List<String> uids) {
+        log.debug("ReportingService:removeUserDetails: uids: {}", uids);
         RemoveUserDetailsParams parameters = new RemoveUserDetailsParams(uids);
         RequestEntity<RemoveUserDetailsParams> requestEntity = requestEntityFactory.createPutRequest(removeUserDetailsUrl, parameters);
         return httpClient.sendRequest(requestEntity, Void.class);
     }
 
-    public ResponseEntity<Void> removeUserDetails(String uid){
-        List<String> uidList = new ArrayList<>(Arrays.asList(uid));
-        return removeUserDetails(uidList);
+    public ResponseEntity<Void> removeUserDetailsDataRetentionJob(List<String> uids) {
+        log.debug("ReportingService:removeUserDetailsDataRetentionJob: uids: {}", uids);
+        RemoveUserDetailsParams parameters = new RemoveUserDetailsParams(uids);
+        RequestEntity<RemoveUserDetailsParams> requestEntity = requestEntityFactory.createPutRequest(removeUserDataFromReportUrl, parameters);
+        return httpClient.sendRequest(requestEntity, Void.class);
     }
 
-
+    public ResponseEntity<Void> removeUserDetails(String uid, Boolean dataRetentionJob) {
```

Comment:
Could we name `dataRetentionJob` to something that makes it clearer that it would run the data retention job when run? Something like `executeDataRetentionJob`? Also, I'm worried about having flag (boolean) arguments as they could potentially be a [code smell](https://martinfowler.com/bliki/FlagArgument.html). I'd recommend extracting the data retention logic from the method.